core/src/main/java/org/jdbi/v3/DBI.java

```diff
@@ -62,28 +65,38 @@
     private final AtomicReference<StatementBuilderFactory> statementBuilderFactory = new AtomicReference<StatementBuilderFactory>(new DefaultStatementBuilderFactory());
     private final AtomicReference<TimingCollector> timingCollector = new AtomicReference<TimingCollector>(TimingCollector.NOP_TIMING_COLLECTOR);
 
+    private final CopyOnWriteArrayList<JdbiPlugin> plugins = new CopyOnWriteArrayList<>();
+
+    private DBI(ConnectionFactory connectionFactory)
+    {
+        if (connectionFactory == null) {
+            throw new IllegalArgumentException("null connectionFactory");
+        }
+        this.connectionFactory = connectionFactory;
+    }
+
     /**
      * Constructor for use with a DataSource which will provide
      *
      * @param dataSource
      */
-    public DBI(DataSource dataSource)
+    public static DBI create(DataSource dataSource)
     {
-        this(dataSource::getConnection);
+        return create(dataSource::getConnection);
     }
 
     /**
-     * Constructor used to allow for obtaining a Connection in a customized manner.
+     * Factory used to allow for obtaining a Connection in a customized manner.
      * <p/>
      * The {@link org.jdbi.v3.tweak.ConnectionFactory#openConnection()} method will
      * be invoked to obtain a connection instance whenever a Handle is opened.
      *
-     * @param connectionFactory PrvidesJDBC connections to Handle instances
+     * @param connectionFactory Prvides JDBC connections to Handle instances
      */
-    public DBI(ConnectionFactory connectionFactory)
-    {
-        assert connectionFactory != null;
-        this.connectionFactory = connectionFactory;
+    public static DBI create(ConnectionFactory connectionFactory) {
+        DBI dbi = new DBI(connectionFactory);
+        dbi.installPlugins();
```

Comment:
This is the only thing I object to. I'm not a fan of Spring-style magic--by which I mean stuff just happens because a JAR is on the classpath (or sometime next year, in the Java 9 module graph).

JDBI has thus far done a great job of keeping the magic "visible." I would rather force users to make a single call to `DBI.installPlugins()` so that the ServiceLoader stuff is truly opt-in.